src/modal/src/components/ModalFooter.js

```diff
@@ -27,24 +24,25 @@ const SecondaryButton :any = styled(Button).attrs(() => ({ mode: 'default' }))`
 `;
 
 type ModalFooterProps = {
-  onClickPrimary ? :() => void;
-  onClickSecondary ? :() => void;
-  shouldStretchButtons ? :boolean;
-  textPrimary ? :string;
-  textSecondary ? :string;
-};
-
-type OverrideFooterProps = {
-  withFooter ? :ComponentType<ModalFooterProps> | boolean;
+  isDisabledPrimary ?:boolean;
+  isDisabledSecondary ?:boolean;
+  isPendingPrimary ?:boolean;
+  isPendingSecondary ?:boolean;
+  onClickPrimary ?:() => void;
+  onClickSecondary ?:() => void;
+  shouldStretchButtons ?:boolean;
+  textPrimary ?:string;
+  textSecondary ?:string;
+  withFooter ?:ComponentType<any> | Element<any> | boolean;
 };
 
-type Props =
-  & ModalFooterProps
-  & OverrideFooterProps;
-
-export default class ModalFooter extends Component<Props> {
+export default class ModalFooter extends Component<ModalFooterProps> {
 
   static propTypes = {
+    isDisabledPrimary: PropTypes.bool,
+    isDisabledSecondary: PropTypes.bool,
+    isPendingPrimary: PropTypes.bool,
+    isPendingSecondary: PropTypes.bool,
```

Comment:
This seems like a slippery slope. Perhaps we should err on the side of leniency with the API and be more strict with internal code reviews?

Reply:
I do agree, this is not ideal. However, `ModalFooter` is an internal component at the moment so this won't affect users.

Reply:
How do you envision this changing if we were to expose the ModalFooter? I have plans to work on different variations of the Modal (ie. `danger`, `warning`, etc.) and it seems like exposing the child components is inevitable.